code/modules/mob/living/status_procs.dm

```diff
@@ -174,6 +174,8 @@
 	. = FALSE
 
 	if(LAZYLEN(sources))
+		if(!islist(sources))
```

Comment:
might want to move the check above the `LAZYLEN` check, it's sort of happening to work because `LAZYLEN` also works on strings.

Reply:
Arguably an empty string should be treated as a null.

Reply:
Actually none of the other trait functions treat empty strings as null so you're right.